src/collectors/collector_worker.go

```diff
@@ -147,6 +151,15 @@ func (cw *CollectorWorker) Run(device configuration.DeviceConfig) error {
 	if err != nil {
 		return err
 	}
+	diff, err := contraGit.GitDiff(*cw.RunConfig, device.Name+".txt", parsed)
```

Comment:
I recognize that a couple lines below this, we pass cw.RunConfig into utils.WriteFile by dereferencing it and copying it by value. I don't think that's necessary in either place, as we're not modifying the values, and needing to retain the original.

Instead of `contraGit.GitDiff(*cw.RunConfig,` just remove the star. `contraGit.GitDiff(cw.RunConfig,` and then in `utils/git/diff.go` for `func GitDiff(config configuration.Config,` add the star to `func GitDiff(config *configuration.Config,`

This way it'll simply pass in the pointer, rather than making a complete in-memory copy of the struct.

Reply:
Furthermore, it looks like the only value that GitDiff needs now is the workspace name. Might be better to pass just workspace, and remove the dependency from `GitDiff` on configuration.Config

Reply:
Updated to pass just the workspace in as a string (both GitDiff and WriteFile)